### PR TITLE
Updated Raku runtime files 

### PR DESCRIPTION
**Pull request by demostanis**
See https://github.com/neovim/neovim/issues/14376 and https://github.com/neovim/neovim/pull/14381.

### CODE REVIEW

**Comment by demostanis** [comment]
What am I supposed to do? (regarding codecov)

**Comment by brammool** [comment]
Comments welcome.  I suppose the "perl6" name was abandoned?
I generally prefer the author or maintainer of the file to request including in the Vim distribution.
That makes sure we have the right version.

File: runtime/ftplugin/raku.vim [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+" Provided by Ned Konz <ned at bike-nomad dot com>
+"---------------------------------------------
+setlocal include=\\<\\(use\\\|require\\)\\>
+setlocal includeexpr=substitute(v:fname,'::','/','g')
+setlocal suffixesadd=.rakumod,.rakudoc,.pm6,.pm
+setlocal define=[^A-Za-z_]
+
+" The following line changes a global variable but is necessary to make
+" gf and similar commands work. Thanks to Andrew Pimlott for pointing out
+" the problem. If this causes a " problem for you, add an
```
Comment (dpelle) [inline runtime/ftplugin/raku.vim]: this causes " a problem → this causes a problem

**Comment by demostanis** [comment]
See https://github.com/Raku/problem-solving/issues/81#issuecomment-519357769 about the name change.
The only changes I've made are to the homepage link:
```
curl https://github.com/Raku/vim-raku/raw/master/syntax/raku.vim -Lso- | diff - runtime/syntax/raku.vim
4,6c4,6
< " Homepage:      https://github.com/vim-perl/vim-perl6
< " Bugs/requests: https://github.com/vim-perl/vim-perl6/issues
< " Last Change:   {{LAST_CHANGE}}
---
> " Homepage:      https://github.com/Raku/vim-raku
> " Bugs/requests: https://github.com/Raku/vim-raku/issues
> " Last Change:   2021-04-16
```
which still point to the old one.
Other than that, it's the same version as upstream.

File: runtime/ftplugin/raku.vim (hunk 1 above)
Comment (demostanis) [reply runtime/ftplugin/raku.vim]: Fixed.

**Comment by demostanis** [comment]
@brammool What's the verdict?
lizmat (someone actively working on Raku) has 👍ed this PR.

**Comment by brammool** [comment]
I suppose perl6 was just a "working title" and no users are expected to edit files with that filetype.  If I understand it correctly Raku is now exclusively used.  So we can drop the "perl6" filetype and just use Raku, right?

**Comment by demostanis** [comment]
Yes, but file extensions .pm6, .p6, .t6 and .pod6 are kept for compatibility purposes.

**Comment by brammool** [comment]
> Yes, but file extensions .pm6, .p6, .t6 and .pod6 are kept for
> compatibility purposes.

And they are marked as Raku, not perl6.  So the "perl6" filetype ceases
to exist.  That looks OK.


-- 
A: Because it messes up the order in which people normally read text.
Q: Why is top-posting such a bad thing?
A: Top-posting.
Q: What is the most annoying thing on usenet and in e-mail?

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///                                                                      \\\
\\\        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by demostanis** [comment]
Exactly. This PR deletes it.

**Comment by b2gills** [comment]
On an unrelated matter: The changes should have been done differently.

The first commit should have only involved renaming the files. That way `git blame` would still work.
The second and subsequent commits would have been the textual changes needed.

(With a small repository you can get away with doing the rename and a small amount of changes in the same commit.)

As you have it, you now get all of the credit/blame for some of the files because you deleted them in one commit, and created them anew under a new name in the next commit.

(There is a way to update this PR to fix these issues, but they involve some git-fu followed by a `git push -‍-‍force`.)

**Comment by demostanis** [comment]
@brammool It looks like you didn't keep the updated runtime files, did you?

**Comment by brammool** [comment]
> @brammool It looks like you didn't keep the updated runtime files, did you?

They will be included in the next runtime file update.  I'll do that
soon, after I have included a few other changes.


-- 
There are three kinds of persons: Those who can count and those who can't.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///                                                                      \\\
\\\        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Pull request closed without merging**